src/optparse.coffee

```diff
@@ -25,37 +25,26 @@ exports.OptionParser = class OptionParser
   # parsers that allow you to attach callback actions for every flag. Instead,
   # you're responsible for interpreting the options object.
   parse: (args) ->
-    options = arguments: []
-    skippingArgument = no
-    originalArgs = args
-    args = normalizeArguments args
-    for arg, i in args
-      if skippingArgument
-        skippingArgument = no
-        continue
-      if arg is '--'
-        pos = originalArgs.indexOf '--'
-        options.arguments = options.arguments.concat originalArgs[(pos + 1)..]
+    state =
+      argsLeft: args[..]
+      options: {}
+    while (arg = state.argsLeft.shift())?
```

Comment:
I’ve been trying to avoid assignments inside conditionals. Can we just use `for arg in args`?